tests/baselines/reference/es6ImportDefaultBindingFollowedWithNamedImportDts.js

```diff
@@ -93,13 +93,13 @@ export declare class x11 {
 }
 //// [client.d.ts]
 import { a } from "./server";
-export declare var x1: a;
 import { a11 as b } from "./server";
-export declare var x2: b;
 import { x, a12 as y } from "./server";
+import { x11 as z } from "./server";

```

Comment:
Again, this is a seemingly unnecessary reordering of imports and exports.

Reply:
IIRC, These are the aliases that aren't actually marked as visible until some time after they are traversed. (If I remember, that ties into the "actually collects declaration errors during the pass instead of after" part of the change) Inserting them in-order would require a bit more work to put them back into the location they were elided from (essentially a n-ary walk just to reinsert imports painted as required during the prior walks) - in the old declaration emitter this was done by tracking the absolute position in the ouput string it was elided from and splicing it in at the end of the walk if it was visible. I guess the similar could be done here (tracking the position in the statement list each import was elided from and the list it was elided from, then rather than inserting at the top, inserting near where the recorded original location was; or maybe substuting an associated transformer-only statement that's been flagged?), but it seems like a lot of extra work for little gain? We don't check TDZ or anything in declaration files, right?

Reply:
Given one of the reasons for .d.ts files is to document the API, it seems like it makes more sense to try to keep the declaration where it originally would have occurred in the file as it is more likely to be logically associated with the export that needed it, resulting in clearer documentation. If .d.ts were ever only intended to be machine readable, I'd be less concerned. I think it's worth preserving the old behavior in this case.

Reply:
Order should now be preserved